src/model/selection.js

```diff
@@ -673,54 +683,27 @@ export default class Selection {
 		for ( const range of this.getRanges() ) {
 			const startBlock = getParentBlock( range.start, visited );
 
-			if ( startBlock ) {
+			if ( startBlock && isTopBlockInRange( startBlock, range ) ) {
```

Comment:
Isn't this also covered by the `for` below?

Reply:
OK, I see why this construct was - to cover collapsed selection and multi-range collapsed selection also (by coincidence I think).